### PR TITLE
Show menues and create protected routes based on beta flag

### PR DESCRIPTION
**Pull request by HaGuesto**
I wrote a simple Authorization hook for routes and some initial new rendering for menues. I also fixed the redirect to dropapp in dev environments to make testing easier.

I updated dev_coordinator, dev_headofops of both organisations in staging and dev tennent in auth0 to have the "beta_user" flag with value 2.

## Therefore to test you should:
- see the menues and be able to access /transfer/agreements route if you are a dev_coordinator
- not see the menues and not be able to access /transfer/agreements route if you are a dev_volunteer

### CODE REVIEW

**Comment by HaGuesto** [comment]
[2.0 Show transfer related menues and create protected routes based on beta flag](https://trello.com/c/Oxbgk0mC/1269-20-show-transfer-related-menues-and-create-protected-routes-based-on-beta-flag)

File: react/src/components/HeaderMenu/HeaderMenuContainer.tsx [hunk 1]
```diff
@@ -102,11 +103,32 @@ const HeaderMenuContainer = () => {
...
 
+  const authorizedMenuItems: MenuItemsGroupData[] = useMemo(() => {
+    return menuItems.filter((menuItem) => {
+      // If no minimum beta requirement exists for the menu item, it should be included.
+      if (!menuItem.minBeta) {
+        return true;
+      }
+
+      // Ensure that auth0.user is defined and parse the beta_user value.
+      if (auth0.user) {
```
Comment (pylipp) [inline react/src/components/HeaderMenu/HeaderMenuContainer.tsx:114]: This is the same logic as line 11ff. in the useAuthorization hook, can it be combined somehow?

File: react/src/hooks/useAuthorization.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const { user } = useAuth0();
+
+  // Check if user has a beta flag higher than the minimum
+  const isAuthorized =
+    user && parseInt(user["https://www.boxtribute.com/beta_user"] ?? "0", 10) >= (minBeta ?? 0);
+
+  // Check if user has all the required permissions
+  //   const isAuthorized = requiredPermissions.every(permission =>
+  //     userPermissions.includes(permission)
+  //   );
```
Comment (pylipp) [inline react/src/hooks/useAuthorization.ts:17]: Delete? :)

**Review by pylipp — APPROVED** [review]
## Functional review
- as dev_coordinator, I see the TRansfers menues, and can go to the /transfers/ pages
- as dev_volunteer, I don't see the transfers menues, and when I go to /transfers/agreements or /shipments, I'm redirected to the QR scanner

## Code review
See comments below. As discussed they will be worked on in a follow-up task because it will include refactoring the menues in general.

**Pull request merged**